egs/librispeech/asr/simple_v1/train.py

```diff
@@ -64,27 +44,33 @@ def get_tot_objf_and_num_frames(tot_scores, frames_per_seq):
     # [ 0 1 3 4 5 ]
     finite_indexes = torch.nonzero(mask).squeeze(1)
     if False:
-        bad_indexes =  torch.nonzero(~mask).squeeze(1)
+        bad_indexes = torch.nonzero(~mask).squeeze(1)
         if bad_indexes.shape[0] > 0:
             print("Bad indexes: ", bad_indexes, ", bad lengths: ", frames_per_seq[bad_indexes],
                   " vs. max length ", torch.max(frames_per_seq), ", avg ",
                   (torch.sum(frames_per_seq) / frames_per_seq.numel()))
-    #print("finite_indexes = ", finite_indexes, ", tot_scores = ", tot_scores)
+    # print("finite_indexes = ", finite_indexes, ", tot_scores = ", tot_scores)
     ok_frames = frames_per_seq[finite_indexes].sum()
     all_frames = frames_per_seq.sum()
     return (tot_scores[finite_indexes].sum(), ok_frames, all_frames)
 
 
-
-def get_objf(batch, model, subsampling, device, L, symbols, training, optimizer=None):
+def get_objf(
+        batch: Dict,
+        model: AcousticModel,
+        device: torch.device,
+        graph_compiler: TrainingGraphCompiler,
+        training: bool,
+        optimizer: Optional[torch.optim.Optimizer] = None
+):
     feature = batch['features']
     supervisions = batch['supervisions']
     supervision_segments = torch.stack(
         (supervisions['sequence_idx'],
-         torch.floor_divide(supervisions['start_frame'], subsampling),
-         torch.floor_divide(supervisions['num_frames'], subsampling)), 1).to(torch.int32)
-    _,indices = torch.sort(-supervision_segments[:,2])
-    supervision_segments = supervision_segments[indices,:]
+         torch.floor_divide(supervisions['start_frame'], model.subsampling_factor),
+         torch.floor_divide(supervisions['num_frames'], model.subsampling_factor)), 1).to(torch.int32)
+    _, indices = torch.sort(-supervision_segments[:, 2])
```

Comment:
I would suggest `torch.argsort` with `descending=True`:
- It returns only the sorted indices, so it does not require to allocate extra memory for the returned tensor.
- It does not require to negate the input.